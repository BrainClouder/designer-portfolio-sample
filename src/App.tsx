import React from 'react';
import { Switch, Route, Redirect } from 'react-router';
import { connect } from 'react-redux';
import Home from './_components/Home';
import { ACTIONS } from './store/actions/main';
import gitDark from './imgs/png/gitHubDark.png';
import gitLight from './imgs/png/gitHubLight.png';
import soLogo from './imgs/svg/so-icon.svg';
import lnLogo from './imgs/png/lnIcon.png';

interface IApp {
  sT: boolean;
  rO: string[];
  aB: boolean;
  sE: boolean;
  styleToggle: () => void;
  aboutToggle: () => void;
}

const App: React.FC<IApp> = ({ aB, sE, sT, rO, styleToggle, aboutToggle }) => {
  const compMap = [<Home />]
  const randomGen = Math.random() < 0.5;
  const profession = Math.random() > 0.5 ? (Math.random() > 0.5 ? (Math.random() > 0.5 ? (Math.random() > 0.5 ? 'UX/UI Designer' : 'Photographer') : 'Art Director') : 'Civil Architect') : 'Designer';
  const highLight = sT ? 'text-orange-400' : 'text-indigo-600';
  return (<>
    <div className={`subpixel-antialiased text-center ${aB ? 'blur-effect' : ''}`}>
      <div className={`m-2 p-2 flex flex-row justify-around ${sE ? 'blur-effect' : ''}`}>
        <div>
          <button onClick={aboutToggle} className={`${sT ? 'text-gray-500 border-gray-800 hover:bg-orange-700' : 'text-gray-700 border-gray-400 hover:bg-indigo-700'} 
            hover:text-white border-2 rounded-full px-4 transform transition duration-200 py-1 
            font-mono text-lg font-bold`}>About</button>
        </div>
        <div>
          <p className={`${highLight} ${sT ? 'logo-effect-dark' : 'logo-effect-light'} font-bold logo-effect`}>can_be_your_logo.com</p>
        </div>
        <label className="label">
          <div className="toggle">
            <input className="toggle-state" type="checkbox" checked={!sT} onChange={styleToggle} />
            <div className="toggle-inner">
              <div className="indicator">
              </div>
            </div>
            <div className="active-bg">
            </div>
          </div>
        </label>
      </div>
      <div>

      </div>
      <div>
        <Switch>
          {rO.map((e: string, i: number) => <Route strict key={e} path={e} component={() => {
            return (
              <div className={`${sT ? 'text-gray-300' : 'text-gray-700'}`}>
                {compMap[i]}
              </div>)
          }
          } />)}
          <Route path={'/'} component={() => <Redirect to={rO[0]} />} />
        </Switch>
      </div>
    </div>

    <div className={`fixed bottom-0 left-0 w-screen`}>
      <div className={`relative mx-auto w-screen text-left`}>
        <div className={`${sT ? 'bg-gray-900' : 'bg-gray-200 shadow-inner'} w-auto inline-block px-2 py-2 m-4 rounded-md`}>
          <div className={`flex flex-col`}>
            <div className="p-2 transition duration-200 transform hover:scale-125">
              <a href={'https://github.com/BrainClouder'}><img src={sT ? gitLight : gitDark} /></a>
            </div>
            <div className="p-2 transition duration-200 transform hover:scale-125">
              <a href={'https://stackoverflow.com/users/11518214/celso-wellington'}><img src={soLogo} width={32}/></a>
            </div>
            <div className="p-2 transition duration-200 transform hover:scale-125">
              <a href={'https://www.linkedin.com/in/contrateme/'}><img src={lnLogo} width={32}/></a>
            </div>
          </div>
        </div>
      </div>
    </div>



    <div className={`${sT ? 'bg-dark' : 'bg-light'}`}></div>
    <div onClick={aboutToggle} className={`fixed top-0 left-0 w-screen bg-black opacity-50 h-screen ${aB ? 'visible' : 'invisible'}`}></div>
    <div className={`absolute top-0 left-0 w-screen ${aB ? 'visible' : 'invisible'}`}>
      <div className={`relative w-full flex flex-row justify-center m-4`}>
        <div onClick={aboutToggle} className={`w-1/4`}></div>
        <div className={`mx-auto m-4 p-2 rounded-lg text-center w-mylg lg:w-mydisplay sm:w-mymd ${sT ? 'bg-gray-900 text-gray-200' : 'bg-gray-200'}`}>
          <div className={`p-2 m-2`}>
            <img className={`rounded-full mx-auto p-2 shadow`} src={`https://randomuser.me/api/portraits/${randomGen ? 'men' : 'women'}/${Math.floor(Math.random() * 68 + 1)}.jpg`}></img>
            <p>{randomGen ? 'Mendonza Schrute' : 'Rachel van der Hopkins'}</p>
            <p className={`font-bold ${highLight}`}>{profession}</p>
            <p>something@email.com</p>
            <p>
              No lies, I'm a proud {profession.toLowerCase()}. Check out my most
              recent works at this page. Talk to me if you interesting in hiring me!
            </p>
          </div>
        </div>
        <div onClick={aboutToggle} className={`w-1/4`}></div>

      </div>
    </div>
  </>
  );
}

const mapStateToProps = (state: any) => {
  const t = state;
  return {
    sT: t.styleToggle,
    rO: t.routes,
    aB: t.aboutState,
    sE: t.selectedArt !== -1,
  }
}

const mapDispatchToProps = (dispatch: any) => {
  return {
    styleToggle: () => dispatch({ type: ACTIONS.nightToggle }),
    aboutToggle: () => dispatch({ type: ACTIONS.toggleAbout })
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
