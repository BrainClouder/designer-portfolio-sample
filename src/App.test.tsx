import React from 'react';
import * as renderer from 'react-test-renderer';
import App from './App';
import { Provider } from 'react-redux';
import store from './store/reducers/main';
import { BrowserRouter as Router } from 'react-router-dom';


// test('renders learn react link', () => {
//   const { getByText } = render(<Provider store={store}>
//     <Router>
//       <App />
//       </Router>
//       </Provider>);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
it("App renders correctly", () => {
  // const props = {
  //   className: "test", value: true, onSelectChanged: (value: boolean) => {return;} };
    const tree = renderer
    .create(<Provider store={store}><Router><App/></Router></Provider>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});