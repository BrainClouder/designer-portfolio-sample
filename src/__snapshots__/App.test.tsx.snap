// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <div
    className="subpixel-antialiased text-center"
  >
    <div>
      Welcome to this designer portfolio! You can select through different modes (if you dont, a random mode will be initialized)
      <button
        className="bg-gray-200 text-gray-900 
            font-bold px-2 py-1 rounded-full"
        onClick={[Function]}
      >
        D
      </button>
    </div>
    <div>
      <div>
        <div
          className="flex flex-wrap justify-center"
        >
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/5106cda8-4842-43ca-a632-6b7c46b6b808/d4rnmov-e5ebdd52-2edf-4c0d-a396-c580f5183920.png/v1/fill/w_894,h_894,q_70,strp/dripping_by_pitel_d4rnmov-pre.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7ImhlaWdodCI6Ijw9MTIwMCIsInBhdGgiOiJcL2ZcLzUxMDZjZGE4LTQ4NDItNDNjYS1hNjMyLTZiN2M0NmI2YjgwOFwvZDRybm1vdi1lNWViZGQ1Mi0yZWRmLTRjMGQtYTM5Ni1jNTgwZjUxODM5MjAucG5nIiwid2lkdGgiOiI8PTEyMDAifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6aW1hZ2Uub3BlcmF0aW9ucyJdfQ.otpnY80z6kw93vZZInW-Gzq97MFsLXuKuEf1F0WdNdY"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "Dripping" by Pitel is licensed under CC BY 3.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://farm2.staticflickr.com/1648/25561336655_5021a21147_b.jpg"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "Superior art" by kevin dooley is licensed under CC BY 2.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://live.staticflickr.com/3805/9019559784_95a6a9a163_b.jpg"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "Boston Museum of Fine Arts" by Si B is licensed under CC BY 2.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://farm9.staticflickr.com/8775/18190488136_3bab22a826.jpg"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "More coffee shop art #art #epfl..." by jeremiah.andrick is licensed under CC BY 2.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://live.staticflickr.com/7226/6862438240_ed622cf929_b.jpg"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "Thi-Thanh-TÃ¢m" by Pascal Maramis is licensed under CC BY 2.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://live.staticflickr.com/7062/7003504291_2f8c8609af_b.jpg"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "House of art" by Janels Katlaps is licensed under CC BY 2.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://live.staticflickr.com/1935/43424378830_079f1af5a1_b.jpg"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "Tempe Center for the Arts" by kevin dooley is licensed under CC BY 2.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://farm9.staticflickr.com/8871/18029751368_4eb3f4aa82.jpg"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "Street art at night Seattle #art..." by jeremiah.andrick is licensed under CC BY 2.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://farm8.staticflickr.com/7010/6607063127_522688ac4c_b.jpg"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "terry_abstract 91_2_72 - Abstract Art Images" by BTerryComptonPhotography is licensed under CC BY-SA 2.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://farm9.staticflickr.com/8524/8525569567_8244e8847b.jpg"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "#wicker #art from #star #design showcased on the wicker furniture blog ." by Wicker Paradise is licensed under CC BY 2.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
          <div
            className="
                    transform transition duration-500 scale-75 rounded-lg p-2
                    cursor-pointer m-2 hover:scale-90
                    "
          >
            <img
              className="mx-auto shadow"
              onClick={[Function]}
              src="https://live.staticflickr.com/5209/5208671435_75cd36711c_b.jpg"
              width={200}
            />
            <div
              className="absolute invisible"
            >
              <div
                className="text-sm p-2 w-auto inline-block"
              >
                "DSC09685" by scalleja is licensed under CC BY-SA 2.0
              </div>
              <button
                className="
                        bg-gray-600 text-white 
                            font-bold px-2 py-1 m-1 rounded-full"
              >
                Learn more
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="bg-dark"
  />,
]
`;
